sessionctx/stmtctx/stmtctx.go

```diff
@@ -134,6 +134,7 @@ type StatementContext struct {
 	NotFillCache     bool
 	MemTracker       *memory.Tracker
 	DiskTracker      *disk.Tracker
+	IsTiFlash        atomic.Value
```

Comment:
maybe using `atomic2.Bool` is better? Then you eve don't need SetIsTiFlash / GetIsTiFlash.